driver-core/src/main/com/mongodb/internal/operation/AsyncSingleBatchCursor.java

```diff
@@ -43,13 +50,12 @@ public void close() {
     public void next(final SingleResultCallback<List<T>> callback) {
         if (closed) {
             callback.onResult(null, new MongoException("next() called after the cursor was closed."));
-        } else if (firstBatch != null && !firstBatch.getResults().isEmpty()) {
-            List<T> results = firstBatch.getResults();
-            firstBatch = null;
-            callback.onResult(results, null);
+        } else if (hasNext && !batch.isEmpty()) {
+            hasNext = false;
+            callback.onResult(batch, null);
         } else {
             closed = true;
-            callback.onResult(null, null);
+            callback.onResult(emptyList(), null);
```

Comment:
Why the change from `null` to `emptyList()` here?  This make it inconsistent with  `AsyncQueryBatchCursor`.

Reply:
So the end state is emptyList for all?

Reply:
This will bring it inline with both sync and async cursors.

Reply:
I peeked ahead at the last PR, and I see some null handling in the sync cursor, e.g.

```
this.nextBatch = commandCursor.getResults().isEmpty() ? null : commandCursor.getResults();
```

Are you looking to change that to allow `nextBatch` to just be set to the empty list if it happens to be empty? 

Relatedly, should we:

* Update the javadoc of `AsyncBatchCursor#next` to reflect this change (iirc you had done that in the previous PR)?
* replace the `results != null` conditional in `BatchCursorFlux#recurseCursor` and the `result == null` check in `BatchCursor#next` with assertions?

Reply:
Added #402 with the javadoc change and recurseCursor change.

`nextBatch` in the sync cursor is a single variable representing a Tenary of values:
 - the nextBatch that can be returned to next
 - if the nextBatch has any results
 - if nextBatch has been processed
 
That said sync cursor `#next()` will never return a null, and now the async cursor won't either.